File: obs-frontend/obs-frontend/src/app/models/book.ts
export interface AuthorRef {
  id: number;
  name: string;
  bio?: string;
}

export interface CategoryRef {
  id: number;
  name: string;
  description?: string;
}

export interface Book {
  id: number;
  title: string;
  description: string;
  price: number;
  isbn: string;
  publishedDate: string;
  authors: AuthorRef[];
  categories: CategoryRef[];
  stock: number;
  coverImageUrl?: string;
}

export interface BookRequest {
  title: string;
  description: string;
  price: number;
  isbn: string;
  publishedDate: string;
  authorIds: number[];
  categoryIds: number[];
  stock: number;
  coverImageUrl?: string;
}

File: obs-backend/src/main/java/com/obs/backend/entity/Book.java
package com.obs.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "books")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false, unique = true, length = 20)
    private String isbn;

    private LocalDate publishedDate;

    @Column(nullable = false)
    private Integer stock;

    @Column(length = 2048)
    private String coverImageUrl;

    @Builder.Default
    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<BookAuthor> bookAuthors = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<BookCategory> bookCategories = new ArrayList<>();
}
